front/app/components/IdeaCards/shared/Filters/TopicFilterDropdown.tsx

```diff
@@ -30,7 +30,10 @@ const TopicFilterDropdown = memo(
     const allowedInputTopics = useProjectAllowedInputTopics(projectId);
 
     const topicIds = getTopicIds(allowedInputTopics);
-    const topics = useTopics({ topicIds });
+    const { data: topics } = useTopics({});
+    const filteredTopics = topics?.data.filter((topic) =>
+      topicIds.includes(topic.id)
```

Comment:
Seeing this pattern repeated quite a few times, if the BE doesn't support an array of topic IDs, maybe we can move this filtering logic into the hook? So you can still do `useTopics({ ids: [...] })`?

Reply:
I was thinking the same thing but I'm hesitant to add it to the hook because it's generally better to return the `useQuery` hook directly to have access to all of the useful things it provides. Otherwise, you need to re-export them which in this case is not a huge deal but as we move forward might be more complicated than keeping the logic in the components themselves. The ideal solution would be to add this functionality to the back-end but for now, I would keep it as is.